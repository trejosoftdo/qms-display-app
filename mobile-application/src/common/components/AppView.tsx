import * as React from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, View } from 'react-native';
import AppLightTheme from '../theme';
import Header from './Header';
import LoadingIndicator from './LoadingIndicator';
import ErrorMessage from './ErrorMessage';


/**
 * AppViewProps defines the props for the App View Component.
 *
 * @property headerMessage - the header display message
 * @property children - the view contained children
 * @property loading - to show the loading indicator
 * @property error - to show a error message
 */
interface AppViewProps {
  headerMessage?: string;
  children: React.ReactNode;
  loading?: boolean;
  error?: Error;
}

/**
 * A component that represents the application general view
 *
 * @param {AppViewProps} props - The props for the App View component.
 */
const AppView: React.FC<AppViewProps> = (props: AppViewProps) => (
  <View style={styles.container}>
    {props.headerMessage && <Header message={props.headerMessage} />}
    {props.loading && <LoadingIndicator loading={props.loading} />}
    {props.children}
    {props.error && <ErrorMessage />}
    <StatusBar style="auto" />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: AppLightTheme.colors.background,
    alignItems: 'center',
    justifyContent: 'flex-start',
    display: 'flex',
    paddingHorizontal: 24,
  },
});

export default AppView;