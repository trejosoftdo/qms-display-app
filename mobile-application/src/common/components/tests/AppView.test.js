import React from 'react';
import AppView from '../AppView';


describe('AppView component', () => {
  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(
      <AppView headerMessage="Test message">
        <div>Test content</div>
      </AppView>
    )).toMatchSnapshot();
  });

  it('renders the loading indicator when loading', () => {
    expect(renderShallow(
      <AppView headerMessage="Test message" loading>
        <div>Test content</div>
      </AppView>
    )).toMatchSnapshot();
  });

  it('renders the error message when there is an error', () => {
    expect(renderShallow(
      <AppView headerMessage="Test message" error>
        <div>Test content</div>
      </AppView>
    )).toMatchSnapshot();
  });
});