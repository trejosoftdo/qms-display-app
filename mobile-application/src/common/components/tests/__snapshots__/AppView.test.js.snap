// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView component renders the component with the provided props as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingHorizontal": 24,
    }
  }
>
  <Header
    message="Test message"
  />
  <div>
    Test content
  </div>
  <ExpoStatusBar
    style="auto"
  />
</View>
`;

exports[`AppView component renders the error message when there is an error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingHorizontal": 24,
    }
  }
>
  <Header
    message="Test message"
  />
  <div>
    Test content
  </div>
  <ErrorMessage />
  <ExpoStatusBar
    style="auto"
  />
</View>
`;

exports[`AppView component renders the loading indicator when loading 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingHorizontal": 24,
    }
  }
>
  <Header
    message="Test message"
  />
  <LoadingIndicator
    loading={true}
  />
  <div>
    Test content
  </div>
  <ExpoStatusBar
    style="auto"
  />
</View>
`;
